shelley/chain-and-ledger/shelley-spec-ledger-test/test/Test/Shelley/Spec/Ledger/Rules/TestChain.hs

```diff
@@ -600,6 +602,28 @@ preserveOutputsTx SourceSignalTarget {source = chainSt, signal = block} =
        in property $
             hasFailedScripts tx || outs `Map.isSubmapOf` u'
 
+canRestrictUTxO ::
+  forall era ledger.
+  ( ChainProperty era,
+    TestingLedger era ledger,
+    HasField "inputs" (Core.TxBody era) (Set (TxIn (Crypto era)))
+  ) =>
+  SourceSignalTarget (CHAIN era) ->
+  Property
+canRestrictUTxO SourceSignalTarget {source = chainSt, signal = block} =
+  conjoin $
+    map outputPreserved $
+      zip (sourceSignalTargets ledgerTrFull) (sourceSignalTargets ledgerTrRestr)
+  where
+    (_, ledgerTrFull) = ledgerTraceFromBlock @era @ledger chainSt block
+    (UTxO irrelevantUTxO, ledgerTrRestr) =
+      ledgerTraceFromBlockWithRestrictedUTxO @era @ledger chainSt block
+    outputPreserved
+      ( SourceSignalTarget {target = (UTxOState {_utxo = (UTxO uFull)}, _), signal = _},
+        SourceSignalTarget {target = (UTxOState {_utxo = (UTxO uRestr)}, _), signal = _}
```

Comment:
I might be wrong, but I don't think you need to bind signal field you don't need it, also redundant parenthesis :
```suggestion
      ( SourceSignalTarget {target = (UTxOState {_utxo = UTxO uFull}, _)},
        SourceSignalTarget {target = (UTxOState {_utxo = UTxO uRestr}, _)}
```